subgraph/src/mappings/vault.ts

```diff
@@ -1,16 +1,88 @@
-import { BigInt, log } from "@graphprotocol/graph-ts";
-import { DepositBurned, DepositMinted } from "../types/templates/Vault/IVault";
+import { BigDecimal, BigInt, ByteArray, log } from "@graphprotocol/graph-ts";
+import {
+  DepositBurned,
+  DepositMinted,
+  YieldClaimed
+} from "../types/templates/Vault/IVault";
 import {
   Sponsored,
   Unsponsored
 } from "../types/templates/Vault/IVaultSponsoring";
-import { Sponsor, Claimer, Deposit, Foundation, Vault } from "../types/schema";
+import {
+  Sponsor,
+  Claimer,
+  Deposit,
+  Foundation,
+  Vault,
+  Donation
+} from "../types/schema";
+
+export function handleYieldClaimed(event: YieldClaimed): void {
+  const claimerId = event.params.claimerId.toString();
+  const claimer = Claimer.load(claimerId)!;
+  const vault = Vault.load(claimer.vault)!;
+
+  claimer.claimed = claimer.claimed.plus(event.params.amount);
+  claimer.shares = claimer.shares.minus(event.params.burnedShares);
+  vault.totalShares = vault.totalShares.minus(event.params.burnedShares);
+  log.debug("claim, sub shares {}", [event.params.burnedShares.toString()]);
+
+  let totalClaimedShares = new BigInt(0);
+
+  for (let i = 0; i < claimer.depositsIds.length; i++) {
+    const deposit = Deposit.load(claimer.depositsIds[i]);
+
+    if (!deposit) continue;
+
+    const claimedShares = deposit.shares
+      .times(event.params.amount)
+      .div(event.params.burnedShares)
+      .minus(deposit.amount)
+      .times(event.params.burnedShares)
+      .div(event.params.amount);
+
+    totalClaimedShares = totalClaimedShares.plus(claimedShares);
+    deposit.shares = deposit.shares.minus(claimedShares);
+    deposit.save();
+
+    if (
+      event.params.to.equals(
+        ByteArray.fromHexString("0x4940c6e628da11ac0bdcf7f82be8579b4696fa33")
```

Comment:
what's this address?

Reply:
Nice catch. I forgot to do this. It needs to come from an event emitted by the vault.